shoes-core/lib/shoes/dimensions.rb

```diff
@@ -82,6 +82,16 @@ def in_bounds?(x, y)
       x_dimension.in_bounds?(x) && y_dimension.in_bounds?(y)
     end
 
+    def contains?(other)
+      return false unless other.element_left && other.element_top &&
+                          other.element_right && other.element_bottom
```

Comment:
shouldn't we always have these?

Reply:
When we're in the midst of laying out the elements (i.e. contents alignment), they haven't necessarily been assigned yet. It's awkward, but definitely necessary with the current system. I've noticed it in other spots too and would like to revise, but it'd probably take some deep 🔪 on the slot placement which I don't have the brain-power for at the moment :)